discord/client.py

```diff
@@ -946,6 +947,39 @@ def event(self, coro):
 
     # Miscellaneous stuff
 
+    async def get_widget(self, guild_id):
+        """|coro|
+
+        Gets a :class:`Widget` from a guild ID.
+
+        .. note::
+
+            The guild must have the widget enabled to get this information.
+
+        Parameters
+        -----------
+        guild_id: :class:`int`
+            The ID of the guild.
+
+        Raises
+        -------
+        Forbidden
+            The widget for this guild is disabled.
+        HTTPException
+            Retrieving the widget failed.
+
+        Returns
+        --------
+        :class:`Widget`
+            The widget.
+        """
+        data = await self.http.get_widget(guild_id)
+        invite = None
+        if data['instant_invite']:
+            invite = await self.get_invite(data['instant_invite'])
```

Comment:
I originally planned to move it to `Widget` itself, but `_resolve_invite` was attached to `Client`, which I felt like would add extra effort just to access it. I feel like if I were to move it to `Widget`, it would be better if `_resolve_invite` was moved to `utils.py`.